src/main/java/com/aws/greengrass/clientdevices/auth/iot/registry/ThingRegistry.java

```diff
@@ -135,12 +139,30 @@ public boolean isThingAttachedToCertificate(Thing thing, Certificate certificate
                 updateThing(thing);
             }
         } catch (CloudServiceInteractionException e) {
-            List<String> certIds = thing.getAttachedCertificateIds();
-            if (certIds.contains(certificate.getCertificateId())) {
+            Map<String, Instant> certIds = thing.getAttachedCertificateIds();
+            if (certIds.containsKey(certificate.getCertificateId())) {
```

Comment:
Following to my comment above this should be something you can live inside of a `Thing`

Reply:
Thing now exposes this method: `Thing.isCertificateAttached(certId)`, cert registry can use that.